src/UI/Art/Animation/MoveAnimation.java

```diff
@@ -14,12 +15,11 @@
   private final GamePath path;
   private final double tilesPerMs;
 
-  public MoveAnimation(int tileSize, Unit actor, GamePath path)
+  public MoveAnimation(int tileSize, GameMap map, Unit actor, GamePath path)
   {
     super(tileSize, actor, null);
     this.path = path;
-    // We don't need an *exact* speed, so this should be fine
-    tilesPerMs = actor.model.getMovePower() / (double) maxTime;
+    tilesPerMs = actor.getMovePower(map) / (double) maxTime;
```

Comment:
This seems a bit unnecessary. Also I still don't know why we need the map to calculate move power.

Reply:
Actually, `UnitContext.calculateMovePower()` does not require the map. Why does this?

Reply:
Gotta nyoom at accurate velocity :v
I have vague plans for a CO that gets bonus movement based on the terrain the unit starts on. I suppose it would be possible to do that via some hacky stuff in `initTurn()`, but it'd be kind of a pain.

Reply:
> Actually, `UnitContext.calculateMovePower()` does not require the map. Why does this?

`UnitContext` has a pointer to the map in it, which will be accessible by the modifiers.